modules/openapi-generator/src/main/resources/typescript-fetch/api.mustache

```diff
@@ -1,364 +1,442 @@
-    {{/bodyParam}}
-
-            return {
-                url: url.format(localVarUrlObj),
-                options: localVarRequestOptions,
-            };
-        },
-    {{/operation}}
-    }
-};
+        }
 
-/**
- * {{classname}} - functional programming interface{{#description}}
- * {{{description}}}{{/description}}
- * @export
- */
-export const {{classname}}Fp = function(configuration?: Configuration) {
-    return {
-    {{#operation}}
-        /**
-         * {{&notes}}
-         {{#summary}}
-         * @summary {{&summary}}
-         {{/summary}}
-         {{#allParams}}
-         * @param {{=<% %>=}}{<%&dataType%>}<%={{ }}=%> {{^required}}[{{/required}}{{paramName}}{{^required}}]{{/required}} {{description}}
-         {{/allParams}}
-         * @param {*} [options] Override http request option.
-         * @throws {RequiredError}
-         */
-        {{nickname}}({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}, {{/allParams}}options?: any): (fetch?: FetchAPI, basePath?: string) => Promise<{{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}Response{{/returnType}}> {
-            const localVarFetchArgs = {{classname}}FetchParamCreator(configuration).{{nickname}}({{#allParams}}{{paramName}}, {{/allParams}}options);
-            return (fetch: FetchAPI = portableFetch, basePath: string = BASE_PATH) => {
-                return fetch(basePath + localVarFetchArgs.url, localVarFetchArgs.options).then((response) => {
-                    if (response.status >= 200 && response.status < 300) {
-                        return response{{#returnType}}.json(){{/returnType}};
-                    } else {
-                        throw response;
-                    }
-                });
-            };
-        },
-    {{/operation}}
+        {{/isListContainer}}
+        {{^isListContainer}}
+        if (requestParameters.{{paramName}} !== undefined) {
+            formData.append('{{baseName}}', requestParameters.{{paramName}} as any);
+        }
+
+        {{/isListContainer}}
+        {{/formParams}}
+        return this.request<{{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}Response{{/returnType}}>({
+            path: `{{{path}}}`{{#pathParams}}.replace(`{${"{{baseName}}"}}`, encodeURIComponent(String(requestParameters.{{paramName}}))){{/pathParams}},
+            method: '{{httpMethod}}',
+            headers: headerParameters,
+            {{#hasQueryParams}}
+            query: queryParameters,
+            {{/hasQueryParams}}
+            {{#hasBodyParam}}
+            {{#bodyParam}}
+            body: requestParameters.{{paramName}},
+            {{/bodyParam}}
+            {{/hasBodyParam}}
+            {{#hasFormParams}}
+            body: formData,
+            {{/hasFormParams}}
+            {{#returnType}}
+            responseType: this.getResponseType('{{{returnType}}}'),
+            {{/returnType}}
+            modelPropertyNaming: '{{modelPropertyNaming}}',
```

Comment:
@TiFu We use the `modelPropertyNaming` flag to do response model property renaming. Currently it uses a runtime method to do the conversion but a better implementation would modify this to use a property name map.

The existing codegen, using the default `camelCase` for model property naming just causes broken code for clients. This is a major barrier for new users IMO when their first experience is a generator results in a broken default.

see: https://github.com/swagger-api/swagger-codegen/issues/3818